components/org.wso2.carbon.identity.verification.onfido.api/org.wso2.carbon.identity.verification.onfido.api.v1/src/main/java/org/wso2/carbon/identity/verification/onfido/api/v1/core/OnfidoIdvService.java

```diff
@@ -0,0 +1,536 @@
+import static org.wso2.carbon.identity.verification.onfido.connector.constants.OnfidoConstants.ONFIDO_LAST_VERIFIED;
+import static org.wso2.carbon.identity.verification.onfido.connector.constants.OnfidoConstants.ONFIDO_VERIFICATION_STATUS;
+import static org.wso2.carbon.identity.verification.onfido.connector.constants.OnfidoConstants.ONFIDO_WORKFLOW_RUN_ID;
+import static org.wso2.carbon.identity.verification.onfido.connector.constants.OnfidoConstants.ONFIDO_WORKFLOW_STATUS;
+import static org.wso2.carbon.identity.verification.onfido.connector.constants.OnfidoConstants.OUTPUT;
+import static org.wso2.carbon.identity.verification.onfido.connector.constants.OnfidoConstants.RESULT;
+import static org.wso2.carbon.identity.verification.onfido.connector.constants.OnfidoConstants.TOKEN;
+import static org.wso2.carbon.identity.verification.onfido.connector.constants.OnfidoConstants.WEBHOOK_TOKEN;
+import static org.wso2.carbon.identity.verification.onfido.connector.constants.OnfidoConstants.WORKFLOW_ID;
+
+/**
+ * Onfido Identity Verification Service implementation to be notified when the verification is completed.
+ */
+public class OnfidoIdvService {
+
+    private static final Log log = LogFactory.getLog(OnfidoIdvService.class);
+
+    /**
+     * Handles the Onfido webhook verification status update.
+     * This method is invoked when Onfido sends a verification status update via webhook.
+     * It validates the request verifying the signature, and updates the verification claims of the user
+     * based on the status.
+     *
+     * @param xSHA2Signature The SHA-2 signature from the Onfido webhook header for validation.
+     * @param idvpId         The identity verification provider ID.
+     * @param verifyRequest  The verification request payload from Onfido.
+     */
+    public void verify(String xSHA2Signature, String idvpId, VerifyRequest verifyRequest) {
+
+        int tenantId = getTenantId();
+        try {
+            String rawRequestBody = RawRequestBodyInterceptor.getRawRequestBody();
+
+            validateResourceTypeAndAction(verifyRequest);
+
+            IdVProvider idVProvider = getIdVProvider(idvpId, tenantId);
+            Map<String, String> idVProviderConfigProperties = getIdVConfigPropertyMap(idVProvider);
+
+            validateSignature(xSHA2Signature, idVProviderConfigProperties, rawRequestBody);
+
+            updateIdVClaims(verifyRequest, idvpId, tenantId, idVProvider);
+        } catch (OnfidoClientException e) {
+            handleClientException(e);
+        } catch (OnfidoServerException e) {
+            handleServerException(e);
+        } finally {
+            RawRequestBodyInterceptor.clear();
+        }
+    }
+
+    /**
+     * Retrieves the tenant ID from the current context.
+     *
+     * @return The tenant ID.
+     * @throws APIError If there is an error retrieving the tenant ID.
+     */
+    private int getTenantId() {
```

Comment:
Shall we move this method to a common util class? Maybe to `org.wso2.carbon.identity.verification.onfido.api.common` package

Reply:
Resolved via https://github.com/wso2-extensions/identity-verification-onfido/pull/12/commits/95c30c53a7755ddb13b54ce67aa26e1abfb47327